### ISSUE TITLE
Tactical Store provides API for recording a mutation

### ISSUE THREAD

**Issue by alxhub** [angular/tactical#32]
The Tactical Store needs an interface to record a mutated version of an object, and should persist it to Idb.

**Comment by ttowncompiled** [comment]
```typescript
/**
 * A storage backend.
 */
export interface Store {
  /**
   * Reads a Record from the store. If a version is supplied, 'fetch' will emit the object
   * associated with that version. Otherwise, 'fetch' will emit the most recently committed
   * version of that object. If the key or version does not match any Record in the store,
   * then 'fetch' will emit null.
   */
  fetch(key: Object, version?: Version): Observable<Record>;

  /**
   * Stores a Record into the store as the most recent version. If the key or version
   * does not match any Record in the store, 'push' will emit false. Otherwise, 'push' will emit
   * true. If the current version has a pending mutation, 'push' will emit the Record for the
   * pending mutation 'onError'.
   */
  push(key: Object, value: Object, base: string): Observable<boolean>;
  
  /**
   * Stores a pending mutation into the store. If the key or version does not match any Record in
   * the store, 'commit' will emit null. If the target version for the mutation is no longer
   * valid, 'commit' will emit null and emit the Record for the valid version 'onError'. Otherwise,
   * 'commit' will emit the Record for the pending mutation.
   */
  commit(key: Object, mutation: Object, version: Version): Observable<Record>;
  
  /**
   * Removes the pending mutation associated with provided key and base. If the provided key or base
   * does not match any Record in the store, then 'rollback' will emit false. Otherwise, 'rollback'
   * will emit true. If the base provided does not match the most recent Record, then the
   * Record matching the base provided will be removed from the store.
   */
  rollback(key: Object, base: string): Observable<boolean>;
  
  /**
   * Removes a Record or RecordChain from the store. If a base is provided, the Record matching
   * the key and base (and the pending mutation to that Record) will be removed from the store.
   * Otherwise, all Records associated with the provided key will be removed from the store. If the
   * key or base does not match any Record in the store, 'remove' will emit false. Otherwise,
   * 'remove' will emit true.
   */
  remove(key: Object, base?: string): Observable<boolean>;
}
```

**Comment by ttowncompiled** [comment]
Closed by PR #46
